marker-api.js

```diff
@@ -1,780 +1,780 @@
+              });
+
+      },
+
+      editMarker: function (req, res) {
+          var tags = req.body.tags,
+              marker_id = req.body.public_id;
+
+          // Only tags are editable for now
+          if (tags) {
+
+              // Update marker data
+              Marker.findOne({'_id': marker_id}, function (err, marker) {
+                  if (err) {
+                      console.log(err);
+                      return res.status(500).json({message: 'An internal error occurred'});
+                  }
+
+                  if (!marker)
+                      return res.status(404).json({message: 'No marker was found with id ' + req.query.marker_id});
+
+                  // Make sure user is authorized to see/edit this marker
+                  if (!marker.user_id.equals(req.user._id)) {
+                      return res.status(403).json({message: 'You are not authorized to view this marker'});
+                  }
+
+                  // Update tags
+                  marker.tags = tags;
+
+                  // Any tag update must be reviewed. Set approved to "pending"/0
+                  marker.approved = 0;
+
+                  marker.save(function (err) {
+                      if (err) {
+                          res.json({message: 'An internal error occurred'});
+                      }
+
+                      res.json({
+                          message: marker._id + ' updated successfully',
+                          new_tags: marker.tags,
+                          approved: 0
+                      });
+                  });
+              }); // end update
+
+          } else {
+              // No tags found
+              res.status(422).json({message: 'No tags received. Only tags are editable'});
+          }
+      },
+
+      deleteMarker: function (req, res) {
+          // No id received 
+          if (!req.query.marker_id) {
+              res.status(422).json({message: 'No id received. Deletion is only possible with an id'});
+          }
+
+          console.log('find one');
+          Marker.findOne({'_id': req.query.marker_id}, function (err, marker) { 
+
+              if (err) {
+                  console.log(err);
+                  return res.status(500).json({message: 'Internal server error'});
+              }
+
+              if (!marker) {
+                  return res.status(404).json({message: 'No marker was found with id ' + req.query.marker_id});
+              }
+
+              // Make sure user is authorized to edit/delete this marker
+              if (!marker.user_id.equals(req.user._id)) {
+                  return res.status(403).json({message: 'You are not authorized to view this marker'});
+              }
+
+              // Move photos to archive
+              ['', '_md', '_sm'].forEach((suffix) => {
+
+                  const photo = marker._id.toString() + suffix + '.jpg';
+                  const existing_path = __dirname + '/public/photos/' + photo;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
                  const existing_path = __dirname + "/public/photos/" + photo;
```